sunpy/instr/goes.py

```diff
@@ -583,16 +699,533 @@ def _goes_get_chianti_em(longflux, temp, satellite=8, abundances="coronal",
                                                 np.max(10**modeltemp)))
 
     # Perform spline fit to model data
     spline = interpolate.splrep(modeltemp, modelflux, s=0)
-    denom = interpolate.splev(np.log10(temp), spline, der=0)
-    em = longflux/denom * 1e55
+    denom = interpolate.splev(np.log10(temp.value), spline, der=0)
+    em = longflux.value/denom * 1e55
+    em = u.Quantity(em, unit='cm**(-3)')
 
     return em
+
+def calculate_radiative_loss_rate(goeslc, force_download=False,
+                                  download_dir=None):
+    """
+    Calculates radiative loss rate from GOES observations.
+
+    This function calculates the radiative loss rate as a function of
+    time of solar soft X-ray-emitting plasma across all wavelengths given a
+    LightCurve object containing GOES data.  The radiative loss rate is
+    determined from the GOES isothermal temperature and volume emission
+    measure as a function of time, as calculated by
+    calculate_temperature_em().  See docstring of that function for more 
+    details.  If the LightCurve object does not contain the temperatures and
+    emission measures, but only contain the GOES fluxes, then the temperature
+    and emission measures are calculated using calculate_temperature_em().
+    The unit of the resulting radiative loss rates is W.  Once
+    the radiative loss rates have been found, they are returned as part of a
+    new LightCurve object also containing the metdata, GOES fluxes and
+    corresponding temperatures and emission measures of the input LightCurve
+    object.
+
+    Parameters
+    ----------
+    goeslc : `sunpy.lightcurve.LightCurve`
+        LightCurve object containing GOES data.  The units of these
+        data MUST be W/m^2 (flux), MK (temperature) and cm^-3
+        (emission measure).  If LightCurve object does not contain
+        temperature and emission measure values, they are calculated from
+        the flux values using calculate_temperature_em().
+
+    force_download : (optional) bool
+        If True, the GOES radiative loss data file is downloaded even if
+        already locally stored. It is important to do this if a new version
+        of the file has been generated due to a new CHIANTI version being
+        released or the launch of new GOES satellites.
+        Default=False
+
+    download_dir : (optional) string
+        The directory to download the GOES radiative loss data file to.
+        Default=SunPy default download directory
+
+    Returns
+    -------
+    lc_new : `sunpy.lightcurve.LightCurve`
+        Contains same metadata and data as input LightCurve with the
+        following additional data columns:
+
+        | lc_new.data.temperature - Array of temperature values [MK]
+        | lc_new.data.em - Array of volume emission measure values [cm**-3]
+        | lc_new.data.rad_loss_rate - radiative loss rate of the coronal soft
+        X-ray-emitting plasma across all wavelengths [W]
```

Comment:
Deep deep in random rst documentation.http://docutils.sourceforge.net/docs/ref/rst/restructuredtext.html#line-blocks